### ISSUE TITLE
T4 Templates Don't Work

### ISSUE THREAD

**Issue by Swellenator** [NSwag/NSwag#1]
Hi

I've had a bit of trouble getting the T4 templates to work. Could you provide a full sample with a TypeScript Generator? When I create a new Text Transform in VS, "SyncTypes.tt", save it, it generates "SyncTypes.cs", which just has a lot of C# code that doesn't compile. 

Here is my code. At a minimum I had to change the references to the NuGet folder, and change the name of the first parameter to controllerClass, it was controllerType "var service = generator.Generate(controllerClass..." 

Anyway, thanks for your code, and excuse my issue, this is the first i've created....

	<#@ template debug="true" hostspecific="true" language="C#" #>

	<#@ assembly name="System.Core" #>
	<#@ assembly name="$(SolutionDir)\packages\NSwag.CodeGeneration.0.11.5770.31403\lib\net45\NJsonSchema.CodeGeneration.dll" #>
	<#@ assembly name="$(SolutionDir)\packages\NJsonSchema.CodeGeneration.1.7.5770.30723\lib\net45\NSwag.CodeGeneration.dll" #>
	<#@ assembly name="$(SolutionDir)\packages\NSwag.Core.0.11.5770.31403\lib\net45\NSwag.Core.dll" #>

	<#@ import namespace="NSwag.CodeGeneration" #>

	<#@ import namespace="System.IO" #>
	<#@ import namespace="System.Reflection" #>
	<#@ import namespace="Microsoft.CSharp" #>
	<#@ import namespace="NSwag.CodeGeneration.ClientGenerators.TypeScript" #>
	<#@ import namespace="NSwag.CodeGeneration.SwaggerGenerators.WebApi" #>
	<#@ output extension=".ts" #>

	<#
		// CONFIGURATION
		var assemblyPath = @"../bin/.WebApi.dll";
		var controllerClass = ".WebApi.Controllers.MobileController";
		var urlTemplate = "api/{controller}/{action}/{id}";
		// -------------

		var fullAssemblyPath = Path.GetFullPath(Path.GetDirectoryName(Host.TemplateFile) + assemblyPath); 

		var generator = new WebApiAssemblyToSwaggerGenerator(assemblyPath);
		var service = generator.Generate(controllerClass, urlTemplate);

		var generatorTS = new SwaggerToTypeScriptGenerator(service);

		var provider = new CSharpCodeProvider();
		generatorTS.Class = provider.CreateEscapedIdentifier(Path.GetFileNameWithoutExtension(Host.TemplateFile));

		var code = generatorTS.GenerateFile();
	#>
	<#= code #>

**Comment by rsuter** [comment]
Did you try out the GUI (NSwagStudio) with the same parameters? Is the output correct there?
Why starts your assembly name with a "." (i.e. ".WebApi.dll")? 
Maybe you should download and reference the tools in NSwag.zip instead of the NuGet package...

**Comment by rsuter** [comment]
It seems that the current release has a bug and T4 generation is not working... fix is coming soon

**Comment by rsuter** [comment]
Ok, I've fixed the issue... please download the latest tools and try again. 
You can find a sample .tt file here https://github.com/NSwag/NSwag/blob/master/src/NSwag.Demo.Client/DataService.tt

**Comment by rsuter** [comment]
Btw: If you use ReSharper, you should check out ForTea for better T4 editing support: https://resharper-plugins.jetbrains.com/packages/JLebosquain.ForTea/

**Comment by Swellenator** [comment]
Hi Again

I have gotten the latest NuGet and tried your example. I had to update the assembly references, but the output attached doesn't seem right. Yep, I have used NSwag Studio and it works fine.

I have ForTea now, it is saying that it cannot resolve symbol "TemplateFile" on "Hosts"?

Looks like you can't attach .cs files! Here is a gist. Am I supposed to execute the TransformText method myself??

https://gist.github.com/Swellenator/ab7e03f0f43f7c2f955b

**Comment by rsuter** [comment]
I don't understand your gist... You have to put the code into a .tt file (create an empty T4 template file in visual studio). 

This .tt file generates a .cs or .ts file as child file... You don't have to create the .cs file yourself..

**Comment by Swellenator** [comment]
I don't understand it either! I can assure you I created a new .tt file in VS and put the code in there, the Gist is the generated file. Could it be that if it doesn't compile properly it just outputs the code that doesn't compile? 

I guess it's just my computer or something :(

![dataservice](https://cloud.githubusercontent.com/assets/14356182/10652489/d21fe364-78a2-11e5-8e71-2e631da75450.png)

**Comment by rsuter** [comment]
This is strange... can you post the content of the DataService.tt file?

**Comment by Swellenator** [comment]
Here you go: 

https://gist.github.com/Swellenator/32a179bfd3dd5a694ad1

**Comment by rsuter** [comment]
It's very strange, are the file properties set as follows?

![capture](https://cloud.githubusercontent.com/assets/2603405/10688566/7ad5b8da-7975-11e5-8b50-d8fec95e390e.PNG)

**Comment by Swellenator** [comment]
Ahhhhhhhhh

The custom tool is "TextTemplatingFilePreprocessor" ! That is because when I went file -> new I chose the first thing that said Text Template, which happens to be "Runtime Text Template" rather than the standard "Text Template" file type. 

So, long story short, it was all my fault - it's now giving me an error generating output about file paths but I should be able to fix that up.

Thanks again!
